app/views/dashboard/_calendar.html.erb

```diff
@@ -4,7 +4,32 @@
 
   <% entries.each do |entry| %>
     <div>
-      <%= entry.mood %>
+      <% if entry.mood %>
+        <button class='emoji-btn' type="button" name="button" data-toggle="modal" data-target="#modal<%= date.strftime("%d") %>"><%= entry.mood %></button>
+      <% end %>
+    </div>
+
+    <div class="modal fade" id="modal<%= date.strftime("%d") %>" tabindex="-1" role="dialog" aria-labelledby="diary" aria-hidden="true">
+      <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h5 class="modal-title" id="diary"> Your Diary Entry on <%= date.strftime("%B %d, %Y") %></h5>
+            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+            </button>
+          </div>
+          <div class="modal-body">
+            <% if entry.entry != '' %>
```

Comment:
Is this the logic to display the diary log?

Reply:
Yep! This turns the emoji into a button and when the user clicks the emoji on the calendar, a pop up box will display with their diary entry on that day. The div with the class modal makes the pop up box